scripts/globals/utils.lua

```diff
@@ -551,3 +551,23 @@ function utils.mobTeleport(mob, hideDuration, pos, disAnim, reapAnim)
         end
     end)
 end
+
+local ffxiRotConversionFactor = 360.0 / 255.0
+
+function utils.ffxiRotToDegrees(ffxiRot)
+    return ffxiRotConversionFactor * ffxiRot
+end
+
+
+function utils.lateralTranslateWithOriginRotation(origin, translation)
```

Comment:
Added a raycast check.

I don't think this is strictly what you wanted. But, I think it works. In terms of pulling this stuff fully in to the core, I'll leave that to someone who is more familiar with our overall navmesh / mob pathing situation.

Reply:
See my writeup here for why raycast doesn't solve this issue (and why its generally pretty awful):
https://github.com/LandSandBoat/server/issues/1470

You're better off making a binding for `zone:isValidPosition(x, y, z)` that hooks up to `navmesh::validPosition`

Reply:
Ah, hmmm okay. It didn't seem like that would actually give me what I wanted. It just seemed like it was valid within the zone, but didn't necessarily preclude something on the other side of a wall or "inside" a column?

Reply:
Okay, bound and switched to validPosition. Though, on the Lua side I named it to "isNavigablePoint" because I feel like it's more expressive (see commit msg).

Left the raycast stuff in there but will drop the commit if you want but I figure "more" bindings can't hurt us.

Reply:
I think I'd rather you drop the raycast binding, since it's a fairly well-known term that doesn't operate in the well-known way.